HowTo release Authorino Operator (#173)
Kuadrant/authorino-operator: RELEASE.md

   * Copy the bundle files from `github.com/kuadrant/authorino-operator/tree/v0.W.Z/bundle`
   * Copy `github.com/kuadrant/authorino-operator/tree/v0.W.Z/bundle.Dockerfile` with the proper fix to the COPY commands
   (i.e. remove /bundle from the paths)

### Publishing the Operator in Kubernetes Community Operators (OperatorHub.io)

1. Open a PR in the [Kubernetes Community Operators repo](https://github.com/k8s-operatorhub/community-operators)
([example](https://github.com/k8s-operatorhub/community-operators/pull/1655) | [docs](https://operatorhub.io/contribute)).

2. The usual steps are the same as for the
[OpenShift Community Operators](https://docs.google.com/document/d/1tLveyv8Zwe0wKyfUTWOlEnFeMB5aVGqIVDUjVYWax0U/edit#heading=h.b5tapxn4sbk5)
hub.

## Notes on Authorino Operator’s automated builds and references to the Authorino manifests and default version

* PRs merged to the main branch of Authorino Operator cause a new image to be built (GH Action) and pushed automatically
to `quay.io/kuadrant/authorino-operator:<git-ref>` – the `quay.io/kuadrant/authorino-operator:latest` tag is also moved
to match the latest <git-ref>.

> * Authorino Operator owns the manifests required by the operator, i.e. Authorino CRD + role definitions, and hosts a copy

Reviewer: ```suggestion
* Authorino Operator owns the manifests generated by the operator itself (i.e. Authorino CRD, RBAC), and stores a copy
```

> of the operand’s manifests (CRD+roles), merged together with the deploy and bundle manifests:

Reviewer: ```suggestion
of the operand’s manifests (AuthConfig CRD, RBAC):
```

>   * Stored in `config/deploy/manifests.yaml` file that gathers resources of both operator (CRD, RBAC, Deployment, Service)
>   and operand (CRD, role definitions)

Reviewer: ```suggestion
  * All resources are added to the bundle manifests (`./bundle`) – for installations managed with [Operator Lifecycle Manager (OLM)](https://olm.operatorframework.io/)
  * A single file manifest is generated at `./config/deploy/manifests.yaml`, gathering the resources to install and deploy the Operator without OLM
```

>     * Committing this file allows to install all the manifests and deploy an instance of the operator from a single

Reviewer: ```suggestion
    * Allows installing (CRDs, RBAC) and deploying (Deployment, Services) an instance of the operator from a single
```

    `kubectl apply` command

>     * Tons of docs in the Authorino repo relying on that

Reviewer: ```suggestion
    * Tons of Authorino docs and examples rely on this
```

  * Make target manifests is used to generate the compiled version of all manifests (operator’s and operand’s).
  It accepts as parameters:
  
    * `VERSION`: sets the version of the manifests of the operator generated by the operator-sdk command and image tag of
    the deployment added to `config/deploy/manifests.yaml` (default: latest)
    * `AUTHORINO_VERSION`: sets the operand’s branch that sources the manifests (CRD, role definitions) that are appended
    to `config/deploy/manifests.yaml` (default: latest)
  

>   * Changes in the operand’s CRD/role definitions out date the _config/deploy/manifests.yaml_ file, thus requiring a PR to

Reviewer: ```suggestion
  * Changes in the operand’s CRD/role definitions out date the `config/deploy/manifests.yaml` file, thus requiring a PR to
```

  the operator repo ([example](https://github.com/Kuadrant/authorino-operator/pull/68))
  
  * A make target `verify-manifests` verifies consistency of the manifests
    * Triggered in the CI tests
    * Causes all PRs in the Operator repo to break whenever there’s a change in the operand’s API committed to the main
    branch → requires immediate update of the copy of operand’s manifests in the main branch of the operator and rebase of all PRs
    
* Apart from the operand’s manifests pinned to a specific or latest version in the copy committed to
`config/deploy/manifests.yaml`, the Operator’s code contains hard-coded references to the default version of the operand
to install:

>   * See `v1beta1.DefaultAuthorinoImage` (`api/v1beta1/authorino_types.go`)

Reviewer: ```suggestion
  * Set at compilation time ([`controllers.DefaultAuthorinoImage`](https://github.com/Kuadrant/authorino-operator/blob/03b42633627337bc7d908e36e579340a0d8e12fb/controllers/constants.go#L95))
  * Configured in the `build.yaml` file ([example](https://github.com/Kuadrant/authorino-operator/blob/release-v0.10.0/build.yaml))
```
